### ISSUE TITLE
Facebook Login and Email

### ISSUE THREAD

**Issue by zanematthew** [zanematthew/zm-ajax-login-register#120]
Facebook Login seems to capture user emails for some users, but not others. Reported here: https://wordpress.org/support/topic/facebook-login-not-picking-up-email-and-weird-username?replies=5#post-7418725

**Comment by zanematthew** [comment]
This is a bug in Facebook. Its documented official in Face Book Developers, https://developers.facebook.com/bugs/298946933534016. 

However a solution to better handle the error will be in place.

**Comment by zanematthew** [comment]
Some possible solutions that can be placed in the plugin include displaying a message if the following are true:

* Primary email is listed.
* Email address is pending (although FB should display a message to the user)
* User is signed up via phone number. If so the email is not requested!

Additionally the users Facebook email can be derived *if* they have a user name, i.e., userName@facebook.com

* [Facebook official bug report on the subject](https://developers.facebook.com/bugs/298946933534016)
* [Facebook user settings](https://www.facebook.com/settings?tab=account&section=email&view)
* [Stackoverflow Discussion](http://stackoverflow.com/questions/16630972/facebook-graph-api-wont-return-email-address)

**Comment by exxed** [comment]
I noticed the same thing but it seems more serious. I used my own email to register to my site. I've used the same various times for other apps and it works fine. It also shows on my facebook apps that my email was captured but my wordpress shows email as empty. I'm attaching an image for the same. 

What's fishy is I can see the Profile as (required) but the email address doesn't show that. See the attached screenshot. 

<img width="510" alt="screen shot 2015-09-18 at 12 29 31 am" src="https://cloud.githubusercontent.com/assets/13815425/9943074/bd203714-5d9c-11e5-811c-d536e10fa16b.png">
